### ISSUE TITLE
Heat Exchanger gets blocked on certain fluid amounts

### ISSUE THREAD

**Issue by LordGolias** [BuildCraft/BuildCraft#4135]
BuildCraft version: 7.99.17
Singleplayer or multiplayer: both

Consider the path 
```
oil (cold) -> Distiller -> Heavy Oil (375 mB) + Fuel Gas (2 B)
```

If the 375mB of Heavy Oil is sent to the Heat Exchanger (HE), the HE it will only consume 370 mB of the fluid. More importantly, the remaining 5mB will be stuck at the entrance of the HE, causing no other kind of liquid to enter until e.g. 365mB of the same liquid arrives to the HE.

IMO, the HE should probably not store liquids on its ends and instead only accept liquids from the input pipe to the output pipe (with cooling/heating). This facilitates using the HE on more complex and fun designs where it is being used to heat different types of liquids during operations.

**Comment by LuigiHutch** [comment]
Duplicate of https://github.com/BuildCraft/BuildCraft/issues/3780

**Comment by LordGolias** [comment]
This is about the heat exchanger. The other is about the distiller.

**Comment by Bigdbigd03** [comment]
This issue looks like a mess. Wow
